resources/aws/auto_scaling_group.go

```diff
@@ -161,3 +163,85 @@ func (asg *AutoScalingGroup) findExisting() (*autoscaling.Group, error) {
 
 	return autoScalingGroups.AutoScalingGroups[0], nil
 }
+
+// FindLegacy returns true if there is an ASG created for the same cluster name
+// but not being part of a CloudFormation stack.
+func (asg *AutoScalingGroup) FindLegacy() (bool, error) {
+	a, err := asg.findExisting()
+	if IsNotFound(err) {
+		return false, nil
+	} else if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	var keyNameFound, keyClusterFound bool
+	for _, td := range a.Tags {
+		if *td.Key == tagKeyName && *td.Value == asg.Name {
+			keyNameFound = true
+		} else if *td.Key == tagKeyCluster && *td.Value == asg.ClusterID {
+			keyClusterFound = true
+		} else if strings.HasPrefix(*td.Key, "aws:cloudformation:") {
+			return false, nil
+		}
+	}
+
+	return keyNameFound && keyClusterFound, nil
+}
+
+// DetachInstances detaches instances from this ASG and returns their details
+func (asg *AutoScalingGroup) DetachInstances() ([]*autoscaling.Instance, error) {
+	autoScalingGroup, err := asg.findExisting()
+	if err != nil {
+		return []*autoscaling.Instance{}, microerror.Mask(err)
+	}
+
+	for _, i := range autoScalingGroup.Instances {
+		_, err = asg.Client.DetachInstances(&autoscaling.DetachInstancesInput{
+			AutoScalingGroupName:           aws.String(asg.Name),
+			InstanceIds:                    []*string{i.InstanceId},
+			ShouldDecrementDesiredCapacity: aws.Bool(false),
+		})
```

Comment:
Note that we get retries for free with the reconciler framework.

Reply:
I'd like to retry this particular line: (199) https://github.com/giantswarm/aws-operator/pull/408/files#diff-1124618ae218f5ff63bf668745931e87R199
Otherwise it's possible that only part of instances.

This is inside for loop. I don't know how the framework could improve situation here.

---

I just noticed `InstanceIds:                    []*string{i.InstanceId},`.  @fgimenez why we can't detach all instances in a single call? That would't solve that problem completely w/o retires.

Reply:
Indeed, the AWS API was complaining when I tried removing all of them at once

Reply:
> I'd like to retry this particular line: (199) https://github.com/giantswarm/aws-operator/pull/408/files#diff-1124618ae218f5ff63bf668745931e87R199

Makes sense, on it

Reply:
Done with simple `backoff.Retry`, pls take a look